cluster/gce/gci/configure-helper.sh

```diff
@@ -83,18 +161,121 @@ function safe-format-and-mount() {
   mount -o discard,defaults "${device}" "${mountpoint}"
 }
 
-# Local ssds, if present, are mounted at /mnt/disks/ssdN.
+# Gets a devices UUID and bind mounts the device to mount location in
+# /mnt/disks/by-id/
+function unique-uuid-bind-mount(){
+  local mountpoint=$1
+  local bindpoint=$2 
+
+  # Mount device to the mountpoint
+  mkdir -p "${bindpoint}"
+  echo "Binding '${mountpoint}' at '${bindpoint}'"
+  mount --bind "${mountpoint}" "${bindpoint}"
+  chmod a+w ${bindpoint}
+}
+
+
+# Mounts, bindmounts, or symlinks depending on the interface and format
+# of the incoming device
+function mount-ext(){
+  local ssd=$1
+  local interface=$2
+  local format=$3  
+
+  # TODO: Handle partitioned disks. Right now this code just ignores partitions
+  if [[ ${format} == "fs" ]]; then
+    if [[ ${interface} == "scsi" ]]; then
+      local scsinum=`echo ${ssd} | sed -e 's/\/dev\/disk\/by-id\/google-local-ssd-\([0-9]*\)/\1/'`
+      local actual_device=$(readlink -f ${ssd} | cut -d '/' -f 3)
+      if [[ -z ${scsinum} ]]; then
+        echo "Failed to get the local disk number for device ${ssd}" >&2
+        exit 2
+      fi
+      local mountpoint="/mnt/disks/ssd${scsinum}"
+    else
+      # This path is required because the existing Google images do not
+      # expose NVMe devices in /dev/disk/by-id so we are using the /dev/nvme instead
+      local nvmenum=`echo ${ssd} | sed -e 's/\/dev\/nvme0n\([0-9]*\)/\1/'`
+      local actual_device=$(echo ${ssd} | cut -d '/' -f 3)
+      if [[ -z ${nvmenum} ]]; then
+        echo "Failed to get the local disk number for device ${ssd}" >&2
+        exit 2
+      fi
+      local mountpoint="/mnt/disks/ssd-nvme${nvmenum}"
+    fi
+
+    # actual_device should be the kernel device name e.g. "sdb", "nvme0n1"
+    if [[ ${actual_device} != sd* ]] && [[ ${actual_device} != nvme* ]]; then
+      echo "'actual_device' is not of the correct format. It must be the kernel name of the device, got ${actual_device} instead" >&2
+      exit 1
+    fi
+
+    safe-format-and-mount ${ssd} ${mountpoint}
+
+    # Trigger udev refresh so that newly formatted devices are propagated in by-uuid
```

Comment:
This section can still go in unique-uuid-bind-mount

Reply:
The udevadm refresh commands must run as a prerequisite to get the real UUID for the bindpoint. This is why I originally did not get the bindpoint outside of the function and pass it in: most of the bindmount specific code ends up having to move into the parent function.

It is a trade-off between a cleaner interface (bindpoint as an argument) or a cleaner abstraction of functionality (bindpoint determined inside of unique-uuid-bind-mount).

A compromise could be passing in the ${actual_device} to a specific unique-uuid-bind-mount function which determines the bindpoint and then calls into a more general bind-mount function with a clean mountpoint,bindpoint interface.